File: src/main/java/com/example/demo/entity/Address.java
package com.example.demo.entity;

import lombok.Data;
import lombok.Getter;
import lombok.Setter;

@Data
@Getter
@Setter
public class Address {
    private String street;
    private String number;
    private String zipCode;
    private String city;
    private String provState;
    private String country;
}

File: reactClientApp/src/Components/Models/Address.js
export default class Address {
    constructor(
        street,
        number,
        zipCode,
        city,
        provState,
        country,
        coordinates) {

        this.street = street;
        this.number = number;
        this.zipCode = zipCode;
        this.city = city;
        this.provState = provState;
        this.country = country;
        this.coordinates = coordinates;
    }
}